src/main/java/com/microsoft/graph/tasks/LargeFileUploadTask.java

```diff
@@ -134,124 +134,89 @@ public CompletableFuture<UploadResult<T>> upload(int maxTries, @Nullable IProgre
                         progress.report(amountUploaded, this.totalUploadLength);
                     }
                     if (result.isUploadSuccessful()) {
-                        return CompletableFuture.completedFuture(result);
+                        return result;
                     }
                 }
-                updateSessionStatus().get();
+                updateSessionStatus();
                 uploadTries += 1;
                 if (uploadTries < maxTries) {
                     TimeUnit.SECONDS.sleep((long) 2 * uploadTries * uploadTries);
                 }
-            } catch (IOException| ExecutionException| ServiceException ex) {
-                CompletableFuture<UploadResult<T>> exceptionalResult = new CompletableFuture<>();
-                exceptionalResult.completeExceptionally(ex);
-                return exceptionalResult;
             }
-
-        }
-        CompletableFuture<UploadResult<T>> exceptionalResult = new CompletableFuture<>();
-        exceptionalResult.completeExceptionally(new CancellationException());
-        return exceptionalResult;
+        throw new CancellationException();
```

Comment:
we should probably throw a more explicit exception saying that we retried the maximum number of times here.